src/main/java/io/jenkins/update_center/ArtifactoryRepositoryImpl.java

```diff
@@ -157,11 +160,26 @@ private void initialize() throws IOException {
             JsonResponse json = JSON.parseObject(body.byteStream(), mediaType == null ? StandardCharsets.UTF_8 : mediaType.charset(), JsonResponse.class);
             json.results.forEach(it -> this.files.put("/" + it.path + "/" + it.name, it));
         }
+        this.poms = this.files.values().stream().filter(it -> it.name.endsWith(".pom")).map(ArtifactoryRepositoryImpl::toGav).filter(Objects::nonNull).collect(Collectors.toSet());
         this.plugins = this.files.values().stream().filter(it -> it.name.endsWith(".hpi") || it.name.endsWith(".jpi")).map(ArtifactoryRepositoryImpl::toGav).filter(Objects::nonNull).collect(Collectors.toSet());
+        removeIf(this.plugins, it -> !this.poms.contains(new ArtifactCoordinates(it.groupId, it.artifactId, it.version, "pom")));
         this.wars = this.files.values().stream().filter(it -> it.name.endsWith(".war")).map(ArtifactoryRepositoryImpl::toGav).collect(Collectors.toSet());
+        removeIf(this.wars, it -> !this.poms.contains(new ArtifactCoordinates(it.groupId, it.artifactId, it.version, "pom")));
         LOGGER.log(Level.INFO, "Initialized " + this.getClass().getName());
     }
 
+    // We cannot use Collection#removeIf because we want to log this
+    private static <E> void removeIf(Collection<E> collection, Predicate<? super E> filter) {
+        final Iterator<E> each = collection.iterator();
+        while (each.hasNext()) {
+            final E current = each.next();
+            if (filter.test(current)) {
+                each.remove();
+                LOGGER.log(Level.INFO, "Removing artifact file without corresponding pom file: " + current);
```

Comment:
```suggestion
                LOGGER.log(Level.WARNING, "Removing artifact file without corresponding pom file: " + current);
```

Reply:
> It might be related to manual uploads and hence security concerns

Anyone entitled to release via m-r-p can choose not to and instead upload manually. No security concern really. (Artifactory offers to generate a POM anyway.)